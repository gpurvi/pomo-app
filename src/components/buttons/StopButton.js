import React from 'react';
import {FontAwesomeIcon} from '@fortawesome/react-fontawesome';
import {faStop} from '@fortawesome/free-solid-svg-icons';
import '../../styles/components/TimerButton.css';

const StopButton = (props) => (
    <i
        onClick={props.onStopHandler}
        className='d-inline-block ml-3 ml-lg-4'
    >
        {props.timerRunning ? (
            // started
            <FontAwesomeIcon size='5x' icon={faStop}/>
        ) : (
            // stopped
            <FontAwesomeIcon
                icon={faStop}
                size='5x'
                className='timer-button--grey'
            />
        )}
    </i>
);

export default StopButton;