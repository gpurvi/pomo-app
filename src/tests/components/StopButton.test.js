import React from 'react';
import {shallow} from 'enzyme';
import StopButton from '../../components/buttons/StopButton';

test('renders StopButton with grey stop image', () => {
    const wrapper = shallow(<StopButton/>);
    expect(wrapper).toMatchSnapshot();
});

test('renders StopButton with black stop image', () => {
    const wrapper = shallow(<StopButton/>);
    wrapper.setProps({timerStarted: true});
    expect(wrapper).toMatchSnapshot();
});

test('when clicked calls onStopHandler', () => {
    const onStopHandler = jest.fn();
    const wrapper = shallow(<StopButton onStopHandler={onStopHandler}/>);
    wrapper.simulate('click');
    expect(onStopHandler).toHaveBeenCalled();
});
