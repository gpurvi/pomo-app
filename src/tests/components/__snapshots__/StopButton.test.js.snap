// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders StopButton with black stop image 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <StopButton
    timerStarted={true}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <svg
        height="48"
        viewBox="0 0 24 24"
        width="48"
        xmlns="http://www.w3.org/2000/svg"
>
        <path
                d="M0 0h24v24H0z"
                fill="none"
        />
        <path
                d="M6 6h12v12H6z"
        />
</svg>,
      "onClick": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <path
            d="M0 0h24v24H0z"
            fill="none"
/>,
          <path
            d="M6 6h12v12H6z"
/>,
        ],
        "height": "48",
        "viewBox": "0 0 24 24",
        "width": "48",
        "xmlns": "http://www.w3.org/2000/svg",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "d": "M0 0h24v24H0z",
            "fill": "none",
          },
          "ref": null,
          "rendered": null,
          "type": "path",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "d": "M6 6h12v12H6z",
          },
          "ref": null,
          "rendered": null,
          "type": "path",
        },
      ],
      "type": "svg",
    },
    "type": "i",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <svg
          height="48"
          viewBox="0 0 24 24"
          width="48"
          xmlns="http://www.w3.org/2000/svg"
>
          <path
                    d="M0 0h24v24H0z"
                    fill="none"
          />
          <path
                    d="M6 6h12v12H6z"
          />
</svg>,
        "onClick": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <path
              d="M0 0h24v24H0z"
              fill="none"
/>,
            <path
              d="M6 6h12v12H6z"
/>,
          ],
          "height": "48",
          "viewBox": "0 0 24 24",
          "width": "48",
          "xmlns": "http://www.w3.org/2000/svg",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "d": "M0 0h24v24H0z",
              "fill": "none",
            },
            "ref": null,
            "rendered": null,
            "type": "path",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "d": "M6 6h12v12H6z",
            },
            "ref": null,
            "rendered": null,
            "type": "path",
          },
        ],
        "type": "svg",
      },
      "type": "i",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`renders StopButton with grey stop image 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <StopButton />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <svg
        height="48"
        viewBox="0 0 24 24"
        width="48"
        xmlns="http://www.w3.org/2000/svg"
>
        <path
                d="M0 0h24v24H0z"
                fill="none"
        />
        <path
                d="M6 6h12v12H6z"
                fill="grey"
        />
</svg>,
      "onClick": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <path
            d="M0 0h24v24H0z"
            fill="none"
/>,
          <path
            d="M6 6h12v12H6z"
            fill="grey"
/>,
        ],
        "height": "48",
        "viewBox": "0 0 24 24",
        "width": "48",
        "xmlns": "http://www.w3.org/2000/svg",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "d": "M0 0h24v24H0z",
            "fill": "none",
          },
          "ref": null,
          "rendered": null,
          "type": "path",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "d": "M6 6h12v12H6z",
            "fill": "grey",
          },
          "ref": null,
          "rendered": null,
          "type": "path",
        },
      ],
      "type": "svg",
    },
    "type": "i",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <svg
          height="48"
          viewBox="0 0 24 24"
          width="48"
          xmlns="http://www.w3.org/2000/svg"
>
          <path
                    d="M0 0h24v24H0z"
                    fill="none"
          />
          <path
                    d="M6 6h12v12H6z"
                    fill="grey"
          />
</svg>,
        "onClick": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <path
              d="M0 0h24v24H0z"
              fill="none"
/>,
            <path
              d="M6 6h12v12H6z"
              fill="grey"
/>,
          ],
          "height": "48",
          "viewBox": "0 0 24 24",
          "width": "48",
          "xmlns": "http://www.w3.org/2000/svg",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "d": "M0 0h24v24H0z",
              "fill": "none",
            },
            "ref": null,
            "rendered": null,
            "type": "path",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "d": "M6 6h12v12H6z",
              "fill": "grey",
            },
            "ref": null,
            "rendered": null,
            "type": "path",
          },
        ],
        "type": "svg",
      },
      "type": "i",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
